### PR TITLE
[FLOC-2330] Release maintenance/flocker 1.0.0pre1/ - in_parallel error handling

### PR DESCRIPTION
**Pull request by exarkun**
Fixes https://clusterhq.atlassian.net/browse/FLOC-2330 (sub-task of https://clusterhq.atlassian.net/browse/FLOC-2327) in the 1.0.0 release branch (which will become 1.0.1).

### CODE REVIEW

**Comment by itamarst** [comment]
Why does the maintenace branch have pre1 in the name? Does that mean it won't get merged forward correctly by Buildbot?

File: flocker/node/test/test_change.py [hunk 1]
```diff
@@ -263,6 +290,40 @@ def test_in_parallel(self):
...
+        (synchronously, not returned as a ``Failure``).
+        """
+        subchanges = [BrokenAction(exception=CustomException())]
+        change = in_parallel(changes=subchanges)
+        result = run_state_change(change, DEPLOYER)
+        failure = self.failureResultOf(result, FirstError)
+        self.assertEqual(failure.value.subFailure.type, CustomException)
+        self.flushLoggedErrors(CustomException)
+
+    def test_changes_run_after_exception(self):
```
Comment (itamarst) [inline flocker/node/test/test_change.py]: `in_parallel` does a sort, so in order to continue testing reliably this relies on relative ordering of ControllableAction and BrokenAction instances, so it's not clear to me test will fail consistently lacking the fix. Maybe ensure `BrokenAction` is smaller than everything?

**Comment by exarkun** [comment]
> Why does the maintenace branch have pre1 in the name? Does that mean it won't get merged forward correctly by Buildbot?

Eugh.  Entirely possible.  I was lazy with tab complete. :grimacing:

**Comment by itamarst** [comment]
Yeah:

```
From https://github.com/ClusterHQ/flocker
 * branch            release/flocker-1.0.0pre1 -> FETCH_HEAD
```

**Comment by exarkun** [comment]
> `in_parallel` does a sort, so in order to continue testing reliably this relies on relative ordering of ControllableAction and BrokenAction instances,

Argh, and I did that.  I'm so bad at programming sometimes.

**Comment by itamarst** [comment]
Thanks, address comments above and then merge.

**Pull request closed without merging**